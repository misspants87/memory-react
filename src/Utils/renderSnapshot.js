import React from 'react';
import renderer from 'react-test-renderer';

const renderSnapshot = (component) => {
  return renderer.create(component).toJSON();
}

export default renderSnapshot;
