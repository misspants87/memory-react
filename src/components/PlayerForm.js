import React from 'react';

//A controlled component is one in which react controls the values of form inputs.
class PlayerForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      playerA: '',
      playerB: '',
      showForm: true,
    }

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  //set the player names
  handleChange(event) {
    //event.target is the input element
    event.target.name==='playerA' ? this.setState({playerA: event.target.value}) : this.setState({playerB: event.target.value});
  }

  //send chosen player names back to Game
  handleSubmit(event) {
    this.props.setPlayerNames(this.state.playerA, this.state.playerB);
    event.preventDefault();
    this.setState({
      showForm: false,
    })
  }

  render() {
    if (this.state.showForm) {
      return (
        <form onSubmit = {this.handleSubmit}>
          <label>
            Player A:
            <input
            type="text"
            name='playerA'
            value = {this.state.playerA}
            onChange = {this.handleChange}/>
          </label>
          <label>
            Player B:
            <input
            type="text"
            name='playerB'
            value = {this.state.playerB}
            onChange = {this.handleChange}/>
          </label>
          <button>
            Submit
          </button>
        </form>
      )
    } else {
      return (
        null
      )
    }

  }
}

export default PlayerForm
