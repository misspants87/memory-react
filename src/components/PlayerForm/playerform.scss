.form-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  justify-content: space-between;

  input {
    margin-left: 5px;
    border-radius: 5px;
    font-family: 'Roboto';
  }

  button {
    // border-radius: 10px;
    height: 30px;
    background-color: #6966ff;
    font-family: 'Roboto';
    font-size: 12pt;
    border: none;
  }

  label {
    color: white;
    font-size: 20pt;
    display: flex;
  }

}
