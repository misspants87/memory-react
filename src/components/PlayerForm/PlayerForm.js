import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import './playerform.scss';


//A controlled component is one in which react controls the values of form inputs.
//I believe it's necessary for this component to be stateful if you want it to be a controlled component.

//If you want to use uncontrolled form (normally best to use controlled), meaning the DOM handles the data from the form, and you need access to the forms data, can use a ref (look it up)
const PlayerForm = props => {
  const [playerA, setPlayerA] = useState('');
  const [playerB, setPlayerB] = useState('');

  // set the player's names
  function handleChange(event) {
    let player = event.target;
    player.name ==='playerA' ? setPlayerA(player.value) : setPlayerB(player.value);
  }

  function handleSubmit(event) {
    props.setPlayerNames(playerA, playerB);
    event.preventDefault();
  }

  useEffect(() => {
    return () => {
      // console.log('component did unmount');
    };
  }, []);

    //setting value on input means its a controlled component, and without an onChange handler, the input will be read only, unless you use a ref.
    //value is what shows as the user types
  return (
      <form className='form-container' onSubmit = {handleSubmit} data-test = "Player Form" >

        <label>
          Player 1:
          <input
          type="text"
          name='playerA'
          value = {playerA}
          placeholder = 'Enter name'
          onChange = {handleChange.bind(this)}
          />
        </label>
        <label>
          Player 2:
          <input
          type="text"
          name='playerB'
          value = {playerB}
          placeholder = 'Enter name'
          onChange = {handleChange.bind(this)}
          />
        </label>
        <button>
          Begin Game!
        </button>

      </form>
  )
}

PlayerForm.propTypes = {
  setPlayerNames: PropTypes.func,
}

export default PlayerForm;
