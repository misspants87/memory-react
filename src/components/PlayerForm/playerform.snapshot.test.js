import React from 'react';
import PlayerForm from './PlayerForm';
import renderSnapshot from '../../Utils/renderSnapshot';

test('should render PlayerForm correctly', () => {
  let snapshot = renderSnapshot(
    <PlayerForm
      setPlayerNames={() => 'Fake fn'}
    />
  )
  expect(snapshot).toMatchSnapshot();
})
