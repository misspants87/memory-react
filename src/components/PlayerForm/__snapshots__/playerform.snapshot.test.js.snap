// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlayerForm correctly 1`] = `
<form
  className="form-container"
  data-test="Player Form"
  onSubmit={[Function]}
>
  <label>
    Player 1:
    <input
      name="playerA"
      onChange={[Function]}
      placeholder="Enter name"
      type="text"
      value=""
    />
  </label>
  <label>
    Player 2:
    <input
      name="playerB"
      onChange={[Function]}
      placeholder="Enter name"
      type="text"
      value=""
    />
  </label>
  <button>
    Begin Game!
  </button>
</form>
`;
